### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

**Comment by dadgar** [comment]
The most recent failure from the TestStateStore_Blocking_Timeout is fixed on master which is why I haven't fixed it in this PR

File: client/driver/docker_test.go [hunk 1]
```diff
@@ -42,17 +43,12 @@ func dockerIsRemote(t *testing.T) bool {
...
-
 // Returns a task with a reserved and dynamic port. The ports are returned
 // respectively.
-func dockerTask() (*structs.Task, int, int) {
-	docker_reserved += 1
-	docker_dynamic += 1
+func dockerTask(t *testing.T) (*structs.Task, int, int) {
+	ports := freeport.GetT(t, 2)
+	docker_reserved := ports[0]
+	docker_dynamic := ports[1]
```
Comment (schmichael) [inline client/driver/docker_test.go]: If you're touching this mind updating the variable naming to match regular Go style?

File: client/driver/raw_exec_test.go [hunk 2]
```diff
@@ -265,7 +265,7 @@ func TestRawExecDriver_HandlerExec(t *testing.T) {
 		Driver: "raw_exec",
 		Config: map[string]interface{}{
 			"command": testtask.Path(),
-			"args":    []string{"sleep", "9000"},
+			"args":    []string{"sleep", "9000s"},
```
Comment (schmichael) [inline client/driver/raw_exec_test.go]: This looks like it was fun to debug. :)

Comment (dadgar) [reply client/driver/raw_exec_test.go]: Took me so long! Literally went home after I found the bug. I was just done haha

**Pull request merged**